### PR TITLE
Contribution to MongoDB topic

### PR DESCRIPTION
**Pull request by Bilalbsd**
I have answered to some MongoDB questions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2029,7 +2041,7 @@ This is where data is stored and also where different processing takes place (e.
 <details>
 <summary>What is a master node?</summary><br><b>
 
-Par of a master node responsibilites:
+Par of a master node responsibilites:
```
Comment (andrzejsydor) [inline README.md:2044]: What is the responsibility o the empty space ` ` here?

Comment (Bilalbsd) [reply README.md:2044]: no idea, maybe a missclick

Comment (Bilalbsd) [reply README.md]: I fixed it

Comment (bregman-arie) [reply README.md]: Can you also fix "Par" to "Part"?

File: README.md [hunk 2]
```diff
@@ -1259,18 +1259,27 @@ as key-value pair, document-oriented, etc.
 
 <details>
 <summary>What is a document? What is a collection?</summary><br><b>
+
+  * A document is a record in MongoDB, which is stored in BSON (Binary JSON) format and is the casic unit of data in MongoDB.
```
Comment (andrzejsydor) [inline README.md:1263]:
```suggestion
  * A document is a record in MongoDB, which is stored in BSON (Binary JSON) format and is the basic unit of data in MongoDB.
```

File: README.md [hunk 3]
```diff
@@ -1259,18 +1259,27 @@ as key-value pair, document-oriented, etc.
...
 <details>
 <summary>What is better? Embedded documents or referenced?</summary><br><b>
+
+  * There is no definitive answer to which is better, it depends on the specific use case and requirements. Some explainations : Embedded documents provide atomic updates, while referenced documents allow for better normalization.
 </b></details>
 
 <details>
 <summary>Have you performed data retrieval optimizations in Mongo? If not, can you think about ways to optimize a slow data retrieval?</summary><br><b>
+
+  * Some ways to optimize data retrieval in MongoDB are : indexing, proper schema design, query optimization and database load balancing.
```
Comment (andrzejsydor) [inline README.md:1282]:
```suggestion
  * Some ways to optimize data retrieval in MongoDB are: indexing, proper schema design, query optimization and database load balancing.
```

File: README.md [hunk 4]
```diff
@@ -1285,6 +1294,9 @@ as key-value pair, document-oriented, etc.
 
 <details>
 <summary>What is the difference between find() and find_one()?</code></summary><br><b>
+	
+  * find() returns all documents that match the query conditions.
```
Comment (andrzejsydor) [inline README.md:1298]:
```suggestion
  * `find()` returns all documents that match the query conditions.
```

**Review by andrzejsydor — CHANGES_REQUESTED** [review]
please recheck the spelling

**Pull request merged**